account_sale_stock_report_non_billed/models/stock_move.py

```diff
@@ -108,29 +108,34 @@ def _set_not_invoiced_values(self, qty_to_invoice, invoiced_qty):
         ) * self.sale_line_id.price_reduce
 
     @api.depends("sale_line_id")
-    @api.depends_context("date_check_invoiced_moves")
+    @api.depends_context("non_billed_date")
     def _compute_not_invoiced_values(self):
         for move in self:
-            if not self.env.context.get("date_check_invoiced_moves"):
+            if not self.env.context.get("non_billed_date") or not self.env.context.get(
+                "non_billed_date_start"
+            ):
                 move.quantity_not_invoiced = 0
                 move.price_not_invoiced = 0
                 continue
-            date_start = self.env.context.get("date_check_invoiced_moves_start", False)
-            date_end = self.env.context.get("date_check_invoiced_moves", False)
-            if date_start:
-                date_start = fields.Date.from_string(date_start)
-            if date_end:
-                date_end = fields.Date.from_string(date_start)
+            date_start = fields.Date.from_string(
+                self.env.context["non_billed_date_start"]
+            )
+            date_end = fields.Date.from_string(self.env.context["non_billed_date"])
             invoices_not_cancel = move.invoice_line_ids.filtered(
                 lambda l: l.move_id.state != "cancel"
             )
             moves_in_date = invoices_not_cancel.mapped("move_line_ids").filtered(
                 lambda m: m.date_done >= date_start and m.date_done <= date_end
             )
+            invoice_date_start = False
+            if self.env.context.get("non_billed_invoice_date_start", False):
+                invoice_date_start = fields.Date.from_string(
+                    self.env.context["non_billed_invoice_date_start"]
+                )
```

Comment:
Since v12, the fields are already date type, so no need of `fields.Date.from_string`

```suggestion
            invoice_date_start = self.env.context.get("non_billed_invoice_date_start", False)
```